compiler/bindings/python/iree/compiler/tools/onnx.py

```diff
@@ -0,0 +1,165 @@
+import os
+import tempfile
+import logging
+import onnx
+
+from dataclasses import dataclass
+from typing import IO, Any, Optional, Union
+from pathlib import Path
+
+from .. import CompilerOptions, InputType, TempFileSaver
+from ..extras import onnx_importer
+from ..ir import Context, StringAttr
+from .binaries import invoke_pipeline
+from .core import build_compile_command_line
+from ...runtime import Config, VmInstance, VmModule, load_vm_module
+
+__all__ = [
+    "compile_saved_model",
+    "compile_model"
+]
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_onnx_model(raw_model: onnx.ModelProto, 
+                    data_prop: bool = True, 
+                    data_dir : IO[bytes] | str | os.PathLike | None = None):
+    # Load the model, with possible external data coming from the default
+    # location, or the location specified on the command line.
+    if data_dir:
+        onnx.load_external_data_for_model(raw_model, str(data_dir))
```

Comment:
We'll want to support keeping large data separate as IREE parameters (https://iree.dev/guides/parameters/).

Sample code for that:
* https://github.com/iree-org/iree-turbine/blob/485bd26bee98dca56e4271d81babf291d1aa4d04/shark_turbine/aot/params.py
* https://github.com/iree-org/iree/blob/main/runtime/bindings/python/tests/io_test.py
* https://github.com/iree-org/iree/blob/main/runtime/bindings/python/tests/io_runtime_test.py

I'd consider that a subtask worth its own standalone PR/commit that should have dedicated unit tests.